Examples/Scripts/FreeSurferPipelineBatch-long.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+function identify_timepoints
+{
+    local subject=$1
+    local tplist=""
+    local tp visit n
+
+    #build the list of timepoints (sessions)
+    n=0
+    for session in ${PossibleVisits[*]}; do
+        tp="${subject}_${session}"
+        if [ -d "$StudyFolder/$tp" ] && ! [[ " ${ExcludeVisits[*]} " =~ [[:space:]]"$tp"[[:space:]] ]]; then
+             if (( n==0 )); then 
+                    tplist="$tp"
+             else
+                    tplist="$tplist,$tp"
+             fi
+        fi
+        ((n++))
+    done
+    echo $tplist
+}
+
+get_usage_and_exit(){
+    echo "usage: "
+    echo "FreeSurferPipelineBatch-long.sh [options]"
+    echo "options:"
+    echo "  --runlocal                      run locally [FALSE]"
+    exit -1
+}
+
+command_line_specified_run_local=FALSE
+while [ -n "$1" ]; do
+    case "$1" in
+        --runlocal) shift; command_line_specified_run_local=TRUE ;;
+        *) shift ;;
+    esac
+done
+
+#################################################################################################
+# General input variables
+##################################################################################################
+#Location of Subject folders (named by subjectID)
+StudyFolder="/my/study/path"
+#The list of subject labels, space separated
+Subjects=(HCA6002236 HCA6002237 HCA6002238)
+#The list of possible visits (aka timepoints, sessions) that each subject may have. Timepoint directories should be named <Subject>_<Visit>.
+PossibleVisits=(V1_MR V2_MR V3_MR V4_MR V5_MR V6_MR V7_MR V8_MR V9_MR V10_MR)
+#The list of possible visits that each subject may have. Timepoint (visit) is expected to be named <Subject>_<Visit>.
+#Actual visits (timepoints) are determined based on existing directories that match the visit name pattern.
+ExcludeVisits=(HCA6002237_V1_MR HCA6002238_V1_MR)
+#Longitudinal template labels, one per each subject.
+Templates=(HCA6002236_V1_V2 HCA6002237_V1_V2 HCA6002238_V1_V2)
+
+EnvironmentScript="<HCPInstallDir>/scripts/SetUpHCPPipeline.sh" #Pipeline environment script
+
+# Requirements for this script
+#  installed versions of: FSL, FreeSurfer, Connectome Workbench (wb_command), gradunwarp (HCP version)
+#  environment: HCPPIPEDIR, FSLDIR, FREESURFER_HOME, CARET7DIR, PATH for gradient_unwarp.py
+
+# If you want to use FreeSurfer 5.3, change the ${queuing_command} line below to use
+# ${HCPPIPEDIR}/FreeSurfer/FreeSurferPipeline-v5.3.0-HCP.sh
+
+#Set up pipeline environment variables and software
+source "$EnvironmentScript"
+
+# Log the originating call
+echo "$@"
+
+#NOTE: syntax for QUEUE has changed compared to earlier pipeline releases,
+#DO NOT include "-q " at the beginning
+#default to no queue, implying run local
+QUEUE="long.q"
+#QUEUE="hcp_priority.q"
+
+#parallel options
+parallel_mode=BUILTIN
+max_jobs=4
```

Comment:
Why is a max of 4 the batch default?  parallel.shlib can detect the number of cores.

Reply:
Parallel.shlib doesn't know how many subject level jobs are run on the node. 
So in scenario 1: subject-level jobs are submitted to separate nodes, detecting # of cores is reasonable
In scenario 2 where multiple per-subject jobs are run on the same machine, such as brainmappers, limiting the # of cores (and number of subject jobs) is needed. 
I guess scenario 1 may be set to default (max_jobs=-1) and I'll just put this comment whenever max_jobs is defined in settings.

Reply:
Generally, the queue submission needs to be tailored to the average execution parallelism of the job, regardless of what method the job uses for parallelism.  A default of automatic on-node parallelism would just mean setting the queue submission in a way that runs fewer subjects per node than the number of cores.  It is okay to have a few more total threads running than physical cores, particularly if the jobs aren't exactly a specific parallelism for their entire length, so in general the control doesn't need to be highly calculated.